### PR TITLE
block weighted fox holder average holdings

### PR DESCRIPTION
**Pull request by asamere**

### CODE REVIEW

File: db/views/fox_transfers_v.sql [hunk 1]
```diff
@@ -0,0 +1,12 @@
+create or replace view fox_transfers_v as (
+    select id,txhash,transfer_to as account,transfer_value as delta, block_number
+    from unique_fox_holders_v holders join transfers on holders.account = transfers.transfer_to
+    where token = '0xc770EEfAd204B5180dF6a14Ee197D99d808ee52d'
```
Comment (0xean) [inline db/views/fox_transfers_v.sql:4]: shouldn't these be lower cased now?

File: db/queries/weighted_deltas.sql [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+             from my_transfers_v
+             where block_number <= 16078485
+             group by account
+             order by block_number)
+            union -- ending balance
+            (select account,
+                    sum(delta),
+                    16298368 as block_number
+             from my_transfers_v
+             where block_number <= 16298368
```
Comment (0xean) [inline db/queries/weighted_deltas.sql:29]: can we pull these from the DB so when we update to the correct snapshots it will flow through?

**Review by 0xean — CHANGES_REQUESTED** [review]
How do you envision we name these tables for the flavors of FOX or is there an easy way to perhaps parameterize the token so we can reuse the same views?

File: db/views/fox_transfers_v.sql (hunk 1 above)
Comment (asamere) [reply db/views/fox_transfers_v.sql:4]: they weren't before I rebased. 😁

File: db/queries/weighted_deltas.sql (hunk 2 above)
Comment (asamere) [reply db/queries/weighted_deltas.sql:29]: yes please

File: db/005_hedgey.sql [hunk 3]
```diff
@@ -1,8 +1,8 @@
-insert into staking_contracts(address, contract_name, chain, genesis_block, height) values 
- ('0x2aa5d15eb36e5960d056e8fea6e7bb3e2a06a351', 'hedgeyNFT', 'ETH', 14496728, 0),
- ('0x2aa5d15eb36e5960d056e8fea6e7bb3e2a06a351', 'hedgeyNFT', 'GNO', 21401247 , 0);
-
+-- insert into staking_contracts(address, contract_name, chain, genesis_block, height) values
```
Comment (0xean) [inline db/005_hedgey.sql:1]: confused by this change.

File: indexer/transfers.go [hunk 4]
```diff
@@ -109,6 +109,7 @@ func (app *IndexerApp) indexTransfersForToken(startBlock uint64, endBlock uint64
 					BlockNumber:  iter.Event.Raw.BlockNumber,
 					TxHash:       iter.Event.Raw.TxHash.String(),
 					TokenAddress: tokenAddress,
+					LogIndex:     uint64(iter.Event.Raw.Index),
```
Comment (0xean) [inline indexer/transfers.go:112]: 👏

File: sample.env [hunk 5]
```diff
@@ -1,7 +1,7 @@
 ETH_RPC="https://mainnet.infura.io/v3/YOURKEY"
 
 FOX_GENESIS_BLOCK=7446830
-FOX_ADDRESS_ETH="0xc770EEfAd204B5180dF6a14Ee197D99d808ee52d"
+FOX_ADDRESS_ETH="0xc770eefad204b5180df6a14ee197d99d808ee52d"
```
Comment (0xean) [inline sample.env:4]: this file Is no longer used I don't think, my bad we should delete it

File: db/views/fox_transfers_v.sql [hunk 6]
```diff
@@ -0,0 +1,12 @@
+create or replace view fox_transfers_v as (
```
Comment (0xean) [inline db/views/fox_transfers_v.sql:1]: still have the open question around naming conventions for this vs other flavors of FOX

**Review by 0xean — CHANGES_REQUESTED** [review]

File: db/views/fox_transfers_v.sql (hunk 6 above)
Comment (asamere) [reply db/views/fox_transfers_v.sql:1]: we don't need that view at all really. that and my_transfers are dev tools. I'll update the main/average ones to include the token. another option would be to wrap it in a function, that would let us:
```
select weighted_avg_balances('0xabc123');
```

File: db/005_hedgey.sql (hunk 3 above)
Comment (asamere) [reply db/005_hedgey.sql:1]: me too. errant paste to wrong window and auto-save? my bad I'll fix.

**Review by 0xean — APPROVED** [review]

File: db/005_hedgey.sql (hunk 3 above)
Comment (asamere) [reply db/005_hedgey.sql:1]: oh now I see. I was hacking the migrations to get them to go down/up with our conflicting 005s so I didn't nuke data and have to import again.

Comment (asamere) [reply db/005_hedgey.sql:1]: think tern refused to execute with an empty `up`, so I stuck `select 1;` there

**Pull request merged**